src/mcts/search.cc

```diff
@@ -621,6 +622,7 @@ void Search::EnsureBestMoveKnown() REQUIRES(nodes_mutex_)
   auto bestmove_edge = temperature
                            ? GetBestRootChildWithTemperature(temperature)
                            : GetBestChildNoTemperature(root_node_, 0);
+  if (randombyp) bestmove_edge = GetRandomChildbyP();
```

Comment:
That's a much better way of doing it. Thank you!

I just didn't consider doing it this way because I felt this feature didn't really need to be all that efficient considering how little it will probably be used, and how little overhead it takes on its own.